ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/StateCachesTest.java

```diff
@@ -0,0 +1,171 @@
+        .isEqualByComparingTo(UInt256.ZERO);
+    assertThat(BeaconStateCache.getSlotCaches(stateWithCaches).getBlockProposerRewards())
+        .isEqualByComparingTo(UInt64.ZERO);
+
+    final BeaconState stateAtSlot3 = spec.processSlots(stateWithCaches, UInt64.valueOf(3));
+
+    assertThat(BeaconStateCache.getSlotCaches(stateAtSlot3).getBlockExecutionValue())
+        .isEqualByComparingTo(UInt256.ZERO);
+    assertThat(BeaconStateCache.getSlotCaches(stateAtSlot3).getBlockProposerRewards())
+        .isEqualByComparingTo(UInt64.ZERO);
+
+    // execution value should not be affected by block processing
+    BeaconStateCache.getSlotCaches(stateAtSlot3).setBlockExecutionValue(UInt256.valueOf(42));
+
+    final SignedBeaconBlock blockAtSlot3 = createWorthyBlock(blockRewardsSource);
+
+    final BeaconState stateAtSlot3WithBlock =
+        spec.atSlot(UInt64.valueOf(3))
+            .getBlockProcessor()
+            .processUnsignedBlock(
+                stateAtSlot3,
+                blockAtSlot3.getMessage(),
+                IndexedAttestationCache.NOOP,
+                BLSSignatureVerifier.NO_OP,
+                Optional.empty());
+
+    if (blockRewardsSource == BlockRewardsSource.NONE) {
+      assertThat(BeaconStateCache.getSlotCaches(stateAtSlot3WithBlock).getBlockProposerRewards())
+          .isEqualByComparingTo(UInt64.ZERO);
+    } else {
+      assertThat(BeaconStateCache.getSlotCaches(stateAtSlot3WithBlock).getBlockProposerRewards())
+          .matches(rewards -> rewards.isGreaterThan(UInt64.ZERO));
+    }
+
+    // execution value should not be affected by block processing
+    assertThat(BeaconStateCache.getSlotCaches(stateAtSlot3WithBlock).getBlockExecutionValue())
+        .isEqualByComparingTo(UInt256.valueOf(42));
+  }
+
+  private SignedBeaconBlock createWorthyBlock(final BlockRewardsSource blockRewardsSource) {
+    final ChainBuilder.BlockOptions blockOptions = ChainBuilder.BlockOptions.create();
+
+    switch (blockRewardsSource) {
+      case ATTESTATION -> {
+        final Attestation attestation =
+            chainBuilder.streamValidAttestationsForBlockAtSlot(3).findFirst().orElseThrow();
+
+        blockOptions.addAttestation(attestation);
+      }
+      case SYNC_COMMITTEE -> {
+        final SyncAggregate syncAggregate =
+            dataStructureUtil.randomSyncAggregate(
+                0, 3, 4, 7, 8, 9, 10, 16, 17, 20, 23, 25, 26, 29, 30);
+        blockOptions.setSyncAggregate(syncAggregate);
+      }
+      case SLASHING -> {
+        final Attestation attestation =
+            chainBuilder.streamValidAttestationsForBlockAtSlot(3).findFirst().orElseThrow();
+        blockOptions.addAttesterSlashing(
+            chainBuilder.createAttesterSlashingForAttestation(attestation, bestBlock));
+      }
+      case NONE -> {}
+    }
+
+    return chainBuilder.generateBlockAtSlot(3, blockOptions).getBlock();
+  }
+
+  private enum BlockRewardsSource {
+    ATTESTATION,
+    SYNC_COMMITTEE,
+    SLASHING,
```

Comment:
we could test both proposer slashing and attestation slashing separately for confidence

Reply:
It would be nice but:
- we don't have a `ProposerShlasingGenerator`
- the both codepaths go into the same place

Reply:
> the both codepaths go into the same place
yeah I know, but it's tests, it's not bad to write them without knowledge of implementation
you never know what will be broken
paths could split in the future or whatever
if it's difficult, ok, let's skip

Reply:
you're right.. let me try to add `ProposerSlashingGenerator`